src/client.js

```diff
@@ -22,6 +31,22 @@ class Client {
         if (typeof payload !== "undefined") Object.assign(data, payload)
         this.socket.write(JSON.stringify(data) + "\n")
     }
+
+    // for proper socket closing
+    close() {
+        this.socket.end()
+    }
+
+    // getconnection establishes connection based on underlying OS
+    getconnection() {
+        if ( os.platform() != "win32" ) {
+            this.socket = net.createConnection(process.argv[1]);
```

Comment:
Why have you transform it to `process.argv[1]`? In the original file it's `process.argv[2]`. And now that we've added another flag, it should be `process.argv[3]`

Reply:
rectified argument number